Sources/NIOHTTP1/NIOHTTPClientUpgradeHandler.swift

```diff
@@ -0,0 +1,434 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2019 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+public typealias NIOHTTPClientUpgradeConfiguration = (upgraders: [NIOHTTPClientProtocolUpgrader], completionHandler: (ChannelHandlerContext) -> Void)
+
+/// Errors that may be raised by the `HTTPClientProtocolUpgrader`.
+public enum NIOHTTPClientUpgradeError: Error {
+    case responseProtocolNotFound
+    case invalidHTTPOrdering
+    case upgraderDeniedUpgrade
+}
+
+/// An object that implements `ProtocolUpgrader` knows how to handle HTTP upgrade to
+/// a specific protocol.
```

Comment:
This doc string needs to be extended to cover the distinction between this protocol and `HTTPServerProtocolUpgrader`.